Mathlib/AlgebraicGeometry/AffineScheme.lean

```diff
@@ -251,12 +263,45 @@ namespace IsAffineOpen
 
 variable {X Y : Scheme.{u}} {U : X.Opens} (hU : IsAffineOpen U) (f : Γ(X, U))
 
+attribute [-simp] eqToHom_op in
+/-- The isomorphism `U ≅ Spec Γ(X, U)` for an affine `U`. -/
+@[simps! hom inv]
+def isoSpec :
+    ↑U ≅ Spec Γ(X, U) :=
+  haveI : IsAffine U := hU
+  U.toScheme.isoSpec ≪≫ Scheme.Spec.mapIso
+    (X.presheaf.mapIso (eqToIso U.openEmbedding_obj_top).op).op
+
+open LocalRing in
+lemma isoSpec_hom_apply (x : U) :
```

Comment:
The general pattern seems to be:
```suggestion
lemma isoSpec_hom_val_base_apply (x : U) :
```
That said, I would be in favor of converting to shortening `val_base_apply` to `apply` everywhere, since this is unambiguous.